sycl/include/sycl/ext/oneapi/matrix/matrix-unified.hpp

```diff
@@ -115,7 +121,10 @@ joint_matrix_fill(Group,
                   const T2 &v) {
 #if defined(__SYCL_DEVICE_ONLY__)
 #if defined(__NVPTX__)
-  res.cuda_impl.wi_marray = v;
+  res.matrix_impl.wi_marray = v;
```

Comment:
Sorry when I mentioned about considering renaming e.g. `cuda_impl` to `matrix_impl` it was for the purpose of using the same code in the cuda/hip impls, since they now both use marrays.
So you'd have to make the marray member of `matrix_impl` the same name in both cases: e.g. swapping to `data` in the cuda case. Then you can do

```
#if defined(__NVPTX__) || defined(__HIP_PLATFORM_AMD_MFMA__)
res.matrix_impl.wi_marray = v;
#endif
```
then you can delete 
```
#elif defined(__HIP_PLATFORM_AMD_MFMA__)
  sycl::ext::oneapi::detail::joint_matrix_apply(res.matrix_impl,
                                                [=](T &value) { value = v; });
```

That is simpler and should be faster for compilation.